### PR TITLE
Send play score

### PR DESCRIPTION
**Pull request by FernandoArevaloE**
* function that allows update score by question.
* added documentation for sending play score
* moved case "Create Player" from position 8 to 12

### CODE REVIEW

File: app/src/service/game.js [hunk 1]
```diff
@@ -41,7 +41,40 @@ const terminate = async (gameId) => {
...
+    if (game.ended) throw new BadRequest('Game already completed');
+
+    if (!game.questions.data.find(q => q.id === questionId)) throw new BadRequest('Invalid question');
+
+    const experience = await experienceService.getExperienceById(game.experience_id);
+
+    const question = experience.questions.data.find(q => q.id === questionId);
+    if (!question) throw new NotFound();
+
+    const scoreForQuestion = question.answer === answer ? 1 : 0;
```
Comment (andymotagl) [inline app/src/service/game.js:60]: This can be just a boolean.

File: app/src/service/game.js [hunk 2]
```diff
@@ -41,7 +41,40 @@ const terminate = async (gameId) => {
...
+
+    if (scoreForQuestion === 0) {
+        return {
+            score: scoreForQuestion
+        };
+    }
```
Comment (andymotagl) [inline app/src/service/game.js:66]: And here it can return an empty object if false.

**Review by andymotagl — COMMENTED** [review]
I guess you can compare the previous branch against this one to avoid show the changes from previous branch here while the latter is merged.

File: app/src/service/game.js (hunk 1 above)
Comment (FernandoArevaloE) [reply app/src/service/game.js]: done

File: app/src/service/game.js (hunk 2 above)
Comment (FernandoArevaloE) [reply app/src/service/game.js]: done

File: app/src/service/game.js [hunk 3]
```diff
@@ -34,7 +34,36 @@ const terminate = async (gameId) => {
...
+const updateScore = async (gameId, questionId, answer) => {
+    if (!gameId || !uuidTool.validate(gameId)) throw new BadRequest('Invalid game id');
+    if (!questionId) throw new BadRequest('Invalid question');
+    if (!answer) throw new BadRequest('Invalid answer');
+
+    const game = await gameDAO.getById(gameId);
+    if (!game) throw new NotFound();
+    if (game.ended) throw new BadRequest('Game already completed');
+
+    if (!game.questions.data.find(q => q.id === questionId)) throw new BadRequest('Invalid question');
```
Comment (andymotagl) [inline app/src/service/game.js:46]: Should not be ?
```javascript
game.questions.find....
```

without the data

File: app/src/service/game.js [hunk 4]
```diff
@@ -34,7 +34,36 @@ const terminate = async (gameId) => {
...
+
+    const experience = await experienceService.getExperienceById(game.experience_id);
+
+    const question = experience.questions.data.find(q => q.id === questionId);
```
Comment (andymotagl) [inline app/src/service/game.js:50]: Same as above

File: app/src/service/game.js (hunk 3 above)
Comment (FernandoArevaloE) [reply app/src/service/game.js:46]: yeah, my bad. I did not see that line

File: app/src/service/game.js (hunk 4 above)
Comment (FernandoArevaloE) [reply app/src/service/game.js:50]: done

File: app/src/service/game.js (hunk 3 above)
Comment (FernandoArevaloE) [reply app/src/service/game.js:46]: done

**Review by andymotagl — APPROVED** [review]

**Pull request merged**